MoPubMediationLog.java

```
@@ -0,0 +1,228 @@
+        Listener, AgentWrapper, Shared
+    }
+
+    /**
+     * Instantiate using this constructor if the mediation adapter class name contains meaningful ad
+     * network and ad format names. Otherwise, call setPreferredNetworkName() and setPreferredAdFormat()
+     * to pass in preferred ad network and ad format names.
+     *
+     * @param className The mediation adapter class name obtained at the time of instantiation
+     */
+    public MoPubMediationLog(String className) {
+        parseSdkAndAdFormat(className);
+    }
+
+    public static void logDataNull(String data) {
+        MoPubLog.i(getPrefix() + data + " is null or invalid. Please check again.");
+    }
+
+    public static void logException(Exception e) {
+        MoPubLog.e(getPrefix() + e.getLocalizedMessage());
+    }
+
+    /**
+     * Use this for all general-purpose logging needs
+     *
+     * @param message The information to be logged
+     */
+    public static void log(String message) {
+        MoPubLog.i(getPrefix() + message);
+    }
+
+    /**
+     * Mediation adapters can pass in a preferred name for their SDK to be logged. This overrides
+     * the name parsed and set in setNetworkName().
+     *
+     * @param network The ad network name that is preferred by the adapter
+     */
+    public static void setPreferredNetworkName(String network) {
+        mAdNetwork = network;
+    }
+
+    /**
+     * Mediation adapters can pass in a preferred ad format name to be logged. This overrides the
+     * name parsed and set in setAdFormat().
+     *
+     * @param format The ad format name that is preferred by the adapter
+     */
+    public static void setPreferredAdFormat(String format) {
+        mAdFormat = format;
+    }
+
+    private static String getNetworkName() {
+        return TextUtils.isEmpty(mAdNetwork) ? Constants.UNKNOWN_AD_NETWORK.toString() : mAdNetwork;
+    }
+
+    private static String getAdFormat() {
```

What are the cases for unknown format?? Would we default to a known value in these cases?

---

So far, none, but I anticipate race conditions where, during transitions, `getAdFormat()` can get called and the ad format has not been set yet (either passed in manually, or via the class name parsing). I did not want to return an empty string or a null here. We shouldn't default to a known value because it will then be misleading. We are defaulting to a constant that says `Unknown ad format`, but we can certainly tweak this string to something friendlier.